server/ntb/publish/ntb_nitf.py

```diff
@@ -73,6 +76,34 @@ def format(self, article, subscriber, codes=None, encoding="us-ascii"):
         except Exception as ex:
             raise FormatterError.nitfFormatterError(ex, subscriber)
 
+    def _populate_metadata(self, article):
+        """
+        For tree type vocabularies add the parent if a child is present
+        """
+        vocabularies = list(superdesk.get_resource_service('vocabularies').get(None, None))
+        fields = {'place': 'place_custom', 'subject': 'subject_custom'}
+        for field in fields:
+            vocabulary = self._get_list_element(vocabularies, '_id', fields[field])
+            if not vocabulary:
+                continue
+            vocabulary_items = vocabulary.get('items', [])
+            field_values = article.get(field, [])
+            for value in list(field_values):
+                if not value.get('parent', None):
+                    continue
+                parent = self._get_list_element(field_values, 'qcode', value['parent'])
+                if not parent:
+                    field_values.append(self._get_list_element(vocabulary_items, 'qcode', value['parent']))
```

Comment:
It's ok, usually is not a good practice to put call functions with many parameters on if condition.